interp/src/debugger/cidb.rs

```diff
@@ -0,0 +1,265 @@
+use super::commands::Command;
+use super::context::DebuggingContext;
+use super::io_utils::Input;
+use crate::environment::InterpreterState;
+use crate::interpreter::{ComponentInterpreter, Interpreter};
+use calyx::ir::{self, RRC};
+
+pub(super) const SPACING: &str = "    ";
+pub struct Debugger<'a> {
+    context: &'a ir::Context,
+    main_component: &'a ir::Component,
+    debugging_ctx: DebuggingContext,
+}
+
+impl<'a> Debugger<'a> {
+    pub fn new(
+        context: &'a ir::Context,
+        main_component: &'a ir::Component,
+    ) -> Self {
+        Self {
+            context,
+            main_component,
+            debugging_ctx: DebuggingContext::default(),
+        }
+    }
+
+    pub fn main_loop(
+        &mut self,
+        env: InterpreterState,
+        pass_through: bool, //flag to just evaluate the debugger version
```

Comment:
Does "just evaluate" means run in normal interpreter mode?

Reply:
Basically, though the important difference is that it does this with the stepper structures, which for the moment are still separate from the main interpreter loop. Practically speaking this means that it runs the control structure through the debugger rather than main interpreter path. This only exists to ensure that both paths behave the same. Ideally we'll eventually merge the two paths and this will be unnecessary.